platforms.go

```diff
@@ -144,18 +144,51 @@ type Matcher interface {
 //
 // Applications should opt to use `Match` over directly parsing specifiers.
 func NewMatcher(platform specs.Platform) Matcher {
-	return newDefaultMatcher(platform)
+	m := &matcher{
+		Platform: Normalize(platform),
+	}
+
+	if platform.OS == "windows" {
+		m.osvM = &windowsVersionMatcher{
+			windowsOSVersion: getWindowsOSVersion(platform.OSVersion),
+		}
+		// In prior versions, on windows, the returned matcher implements a
+		// MatchComprarer interface.
+		// This preserves that behavior for backwards compatibility.
+		//
+		// TODO: This isn't actually used in this package at all, which may have been
+		// an unintended side of some refactor.
+		// I suspect that it was intended to be used in `Ordered` but it is not since
```

Comment:
If you say "I" you gotta attribute the TODO or generalize the comment here

Reply:
Updated this.